mono/metadata/handle.h

```diff
@@ -132,7 +150,10 @@ mono_handle_domain (MonoHandle handle)
 #define MONO_HANDLE_TYPE_DECL(type)      typedef struct { type *__private_obj; } type ## HandleStorage ; \
 	typedef type ## HandleStorage * type ## Handle
 #define MONO_HANDLE_TYPE(type)           type ## Handle
-#define MONO_HANDLE_NEW(type,obj)        ((type ## Handle) mono_handle_new ((MonoObject*) (obj)))
+#define MONO_HANDLE_NEW(type,obj)        ((MONO_HANDLE_TYPE(type)) mono_handle_new ((MonoObject*) (obj)))
+
+#define MONO_HANDLE_DCL(type,name)	MONO_HANDLE_TYPE(type) name = MONO_HANDLE_NEW(type,name ## _raw)
```

Comment:
This is fine as it's to be used on icalls, where we dictate a simple convention.

The _raw prefix has the advantage of make it weird to use.